pkg/kubectl/cmd/util/factory.go

```diff
@@ -1224,3 +1224,20 @@ func (f *Factory) NewBuilder(thirdPartyDiscovery bool) *resource.Builder {
 
 	return resource.NewBuilder(mapper, typer, resource.ClientMapperFunc(f.ClientForMapping), f.Decoder(true))
 }
+
+func (f *Factory) ResourcesWithPodTemplates() []*meta.RESTMapping {
```

Comment:
Should align with `UpdatePodSpecForObject` - so `ResourcesWithPodSpecs` probably.